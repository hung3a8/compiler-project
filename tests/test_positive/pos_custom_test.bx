exception DivByZero;

def div(x : int, y : int) : int raises DivByZero {
    if (y == 0) {
        raise DivByZero; 
    } else {
        return x / y; 
    }
}

def main() {
    var x = 1 : int; 
    var ans = 1 : int; 
    while (x >= 0) {
        print(ans); 
        try {
            ans = ans * div(2, x); 
        } except DivByZero {
            print(123); 
        }
        x = x - 1;
        print(ans); 
    }
    print(ans);
}
