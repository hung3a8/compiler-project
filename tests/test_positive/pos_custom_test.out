1
2
2
123
2
2